docs/DeveloperGuide.md

```diff
@@ -4,31 +4,95 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+The app was mainly inspired by [addressbook-level2](https://github.com/se-edu/addressbook-level2).
+
+### Design
+The App consists of a few main components.
+
+* Ui: The UI that deals with getting user input and displaying output.
+* Command: Command to be executed.
+* Parser: Parse user input and make sense of it.
+* Storage: Reads data from, and writes data to, the hard disk.
+* Rex: The main class that initialize the app and starts running.
+
+The rest are just objects that the app requires.
+
+The diagram below shows the basic design.
+![image info](./pictures/UML.png)
+
+The complete class diagram is shown below.
+![image info](./pictures/classDiagram.PNG)
+
+### Implementation
+#### Retrieve Command
+The flow for the retrieve command is as follows:
+![image info](./pictures/retrieveCommandSD.png)
```

Comment:
Should the RetrievePatientCommand object be deleted after finishing the command? I think there are similar problems in the following commands.